core/src/avm2/activation.rs

```diff
@@ -999,9 +1000,28 @@ impl<'a, 'gc, 'gc_context> Activation<'a, 'gc, 'gc_context> {
                 _ => self.unknown_op(op),
             };
 
-            if let Err(e) = result {
-                log::error!("AVM2 error: {}", e);
-                return Err(e);
+            if let Err(error) = result {
+                if let Some(body) = method.body() {
+                    for e in body.exceptions.iter() {
```

Comment:
Can you add a FIXME or TODO explaining that we eventually need to check all methods in the call stack?

Reply:
From my understanding I dont think we need to do that because the error should propagate to each method in the call stack.

Reply:
Oh, I see - that's very clever! @dowgird Can you add an explicit test for this - something like:

```
function triggerException() {
// Do something
}

function indrect() {
	try {
		triggerException();
	} catch (foo} {
		...
	}
}
```

Reply:
Done.